smart_contracts/contracts/test/add-gas-subcall/src/main.rs

```diff
@@ -10,33 +10,33 @@ use casper_contract::contract_api::{runtime, storage};
 
 use casper_types::{
     runtime_args, ApiError, CLType, ContractHash, ContractVersion, EntryPoint, EntryPointAccess,
-    EntryPointType, EntryPoints, Parameter, RuntimeArgs,
+    EntryPointType, EntryPoints, Key, Parameter, RuntimeArgs,
 };
 
-// This is making use of the undocumented "FFI" function `gas()` which is used by the Wasm
-// interpreter to charge gas for upcoming interpreted instructions.  For further info on this, see
-// https://docs.rs/pwasm-utils/0.12.0/pwasm_utils/fn.inject_gas_counter.html
-mod unsafe_ffi {
-    extern "C" {
-        pub fn gas(amount: i32);
-    }
-}
-
-fn safe_gas(amount: i32) {
-    unsafe { unsafe_ffi::gas(amount) }
-}
-
 const SUBCALL_NAME: &str = "add_gas";
+const DATA_KEY: &str = "data";
 const ADD_GAS_FROM_SESSION: &str = "add-gas-from-session";
 const ADD_GAS_VIA_SUBCALL: &str = "add-gas-via-subcall";
 
 const ARG_GAS_AMOUNT: &str = "gas_amount";
 const ARG_METHOD_NAME: &str = "method_name";
 
+fn consume_gas(amount: i32) {
```

Comment:
Would be nice to have a comment on this method. Are you sure it will consume _exactly `amount`_ of gas?

Also, why is `amount` a signed integer, can't it be unsinged?

Reply:
Updated in 6169414 to better reflect what it does